test/e2e/framework/statefulset_utils.go

```diff
@@ -193,11 +194,30 @@ func getStatefulSetPodNameAtIndex(index int, ss *apps.StatefulSet) string {
 	return fmt.Sprintf("%v-%v", ss.Name, index)
 }
 
+// scaleFromStatefulSet returns a scale subresource for a statefulset.
+func scaleFromStatefulSet(ss *apps.StatefulSet, count int32) (*extensions.Scale, error) {
+	return &extensions.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ss.Name,
+			Namespace: ss.Namespace,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: count,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: ss.Status.Replicas,
+			Selector: ss.Spec.Selector,
+		},
+	}, nil
+}
+
 // Scale scales ss to count replicas.
 func (s *StatefulSetTester) Scale(ss *apps.StatefulSet, count int32) error {
 	name := ss.Name
 	ns := ss.Namespace
-	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
+	newScale, _ := scaleFromStatefulSet(ss, count)
+	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = newScale.Spec.Replicas })
```

Comment:
The way to call an endpoint is something like this:
```go
// c is a clientset
// scale is a scale object
c.AppsV1beta2().StatefulSets(ns).Scale(scale)
```
Of course, this needs apps/v1beta2 to be enabled. 


You can take a look at how deployment's rollback endpoint is used:
https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/test/e2e/workload/deployment.go#L676